src/main/java/org/opensrp/web/rest/ClientFormResource.java

```diff
@@ -141,6 +151,42 @@ private ClientFormMetadata getMostRecentVersion(@NonNull String formIdentifier)
         }
     }
 
+    @RequestMapping(method = RequestMethod.GET, path = "release-related-files")
+    private ResponseEntity<String> getAllFilesRelatedToRelease(@RequestParam(value = "identifier") String releaseIdentifier)
+            throws JsonProcessingException {
+        if (StringUtils.isBlank(releaseIdentifier)) {
+            return new ResponseEntity<>("Request parameter cannot be empty", HttpStatus.BAD_REQUEST);
+        }
+
+        List<ClientFormMetadata> clientFormMetadataList = new ArrayList<>();
+
+        Manifest manifest = manifestService.getManifest(releaseIdentifier);
+        if (manifest != null && StringUtils.isNotBlank(manifest.getJson())) {
+            JSONObject json = new JSONObject(manifest.getJson());
+            if (json.has(FORM_IDENTIFIERS)) {
+                JSONArray fileIdentifiers = json.getJSONArray(FORM_IDENTIFIERS);
+                if (fileIdentifiers != null && fileIdentifiers.length() > 0) {
+                    for (int i = 0; i < fileIdentifiers.length(); i++) {
+                        String fileIdentifier = fileIdentifiers.getString(i);
+                        ClientFormMetadata clientFormMetadata = getMostRecentVersion(fileIdentifier);
+                        clientFormMetadataList.add(clientFormMetadata);
+                    }
+                } else {
+                    return new ResponseEntity<>("This manifest does not have any files related to it",
+                                HttpStatus.NO_CONTENT);
+                }
+            } else {
+                return new ResponseEntity<>("This manifest does not have any files related to it",
+                       HttpStatus.NO_CONTENT);
+            }
+        } else {
+            return new ResponseEntity<>("No manifest matching the given identifier was found",
```

Comment:
A bit misleading, just add the ` OR manifest does not have any files related to it`